packages/gestalt/src/Typeahead.js

```diff
@@ -0,0 +1,351 @@
+// @flow strict
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import InputField from './TypeaheadInputField.js';
+import Option from './TypeaheadOption.js';
+import Box from './Box.js';
+import Text from './Text.js';
+import Flyout from './Flyout.js';
+import Layer from './Layer.js';
+
+type TextColors =
+  | 'green'
+  | 'pine'
+  | 'olive'
+  | 'blue'
+  | 'navy'
+  | 'midnight'
+  | 'purple'
+  | 'orchid'
+  | 'eggplant'
+  | 'maroon'
+  | 'watermelon'
+  | 'orange'
+  | 'darkGray'
+  | 'gray'
+  | 'lightGray'
+  | 'red'
+  | 'white';
+
+type OptionObject = {
+  label: string,
+  value: string,
+};
+
+type Props = {|
+  accessibilityLabel: string,
+  id: string,
+  onChange: ({ event: SyntheticInputEvent<>, value: string }) => void,
+  data: Array<{
+    label: string,
+    value: string,
+  }>,
+  placeholder?: string,
+  size?: 'md' | 'lg',
+  value?: ?string,
+  searchField?: string,
+  onBlur?: () => void,
+  onChange?: ({
+    event: SyntheticEvent<HTMLInputElement>,
+    value: string,
+  }) => void => void,
+  onFocus?: () => void,
+  onSelect?: OptionObject => void,
+  value?: string,
+  caret?: boolean,
+  defaultItem?: OptionObject,
+  noResultText?: string,
+  noResultTextColor?: TextColors,
+  resultHeight?: string,
+  hoverColor?: TextColors,
+  textColor?: TextColors,
+  backgroundColor?: TextColors,
+|};
+
+const Typeahead = (props: Props) => {
+  const {
+    accessibilityLabel = 'Demo Search Field',
+    id,
+    onBlur,
+    onChange,
+    onFocus,
+    onSelect,
+    value = '',
+    data,
+    placeholder,
+    size,
+    resultHeight = '50vh',
+    searchField = 'label',
+    defaultItem = null,
+    caret = false,
+    noResultText = 'No Results',
```

Comment:
@simkessy this should be a mandatory prop and we should not have a default. That way we enforce `i18n` for this label.